assets/wizards/engagement/components/prompt.tsx

```diff
@@ -0,0 +1,303 @@
+/* eslint-disable no-nested-ternary */
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { BaseControl, CheckboxControl, TextareaControl } from '@wordpress/components';
+import apiFetch from '@wordpress/api-fetch';
+import { Fragment, useEffect, useState } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import { stringify } from 'qs';
+
+/**
+ * Internal dependencies
+ */
+import {
+	Attachment,
+	InputField,
+	InputValues,
+	PromptOptions,
+	PromptProps,
+	PromptType,
+} from './types';
+import {
+	ActionCard,
+	Button,
+	Grid,
+	ImageUpload,
+	Notice,
+	TextControl,
+	WebPreview,
+} from '../../../components/src';
+
+// Note: Schema and types for the `prompt` prop is defined in Newspack Campaigns: https://github.com/Automattic/newspack-popups/blob/master/includes/schemas/class-prompts.php
+export default function Prompt( { inFlight, prompt, setInFlight, setPrompts }: PromptProps ) {
+	const [ values, setValues ] = useState< InputValues | Record< string, never > >( {} );
+	const [ isDirty, setIsDirty ] = useState( false );
+	const [ error, setError ] = useState< boolean | { message: string } >( false );
```

Comment:
```suggestion
	const [ error, setError ] = useState< false | { message: string } >( false );
```

Will this ever be `true`? Also, my linter was accusing error when reading `message`. Maybe it doesn't support a `type | value` comparison? Changing to `value | value` fixed it.